patches/minecraft/net/minecraft/tileentity/TileEntityBrewingStand.java.patch

```diff
@@ -1,14 +1,27 @@
 --- ../src-base/minecraft/net/minecraft/tileentity/TileEntityBrewingStand.java
 +++ ../src-work/minecraft/net/minecraft/tileentity/TileEntityBrewingStand.java
-@@ -153,6 +153,7 @@
+@@ -72,9 +72,10 @@
+     {
+         ItemStack itemstack = this.field_145945_j.get(4);
+ 
+-        if (this.field_184278_m <= 0 && itemstack.func_77973_b() == Items.field_151065_br)
++        int fuel = net.minecraftforge.event.ForgeEventFactory.getBrewingStandFuel(itemstack);
++        if ((this.field_184278_m <= 0 || this.field_184278_m + fuel <= 20) && fuel > 0)
```

Comment:
`this.field_184278_m <= 0` is redundant here, 
it is handled as part of `this.field_184278_m + fuel <= 20`

Reply:
It's for fuel with values over 20

Reply:
Ah ok.

Reply:
Ya this is just logically wrong.
First its redundant. Secondly it changes the behavior. Just leave the <=0 so it'll only eat fuel when its out of fuel like it normally does.
On top of that if it IS a event you should only fire it when you need to. So move the event into the handler.

Reply:
You're still eating fuel at the top of the handler instead of the other way around. Vanilla eats fuel at zero, that's it.